### PR TITLE
[swift] Properly disambiguate OneOf enum if it has the same name as enclosing type.

### PR DESCRIPTION
**Pull request by dnkoutso**
If a `oneof` field is named the same as one of the chain of messages it belongs to and it also declares a message type as an one its options, then it can cause ambiguous Swift compile errors when it comes to scoping due to the way code is emitted.

This change fixes this edge case by scanning the chain of message types and names if they completely match the name of the `oneof` declaration and appends `_OneOf` to the type to help disambiguate.

A test was expanded that fails without this change and passes with it and also adds more edge case scenarios to verify it works.

### CODE REVIEW

File: wire-tests/src/commonTest/proto/kotlin/swift_edge_cases.proto [hunk 1]
```diff
@@ -37,6 +37,11 @@ message Screen {
   oneof screen {
     string oneof_string = 1;
     int32 oneof_int32 = 2;
+    SubMessage oneof_sub_message = 3;
```
Comment (dnkoutso) [inline wire-tests/src/commonTest/proto/kotlin/swift_edge_cases.proto:40]: Compiling generated Swift code without the change in the SwiftGenerator above fails. 

Also verified in our internal repo this fixes this edge case.

Comment (dnkoutso) [reply wire-tests/src/commonTest/proto/kotlin/swift_edge_cases.proto:40]: Also I should note there are other protos that do not have the same name as the enclosing message name that from this change do not use the `_OneOf` suffix. This suffix is only used in this edge case.

File: wire-tests/src/commonTest/proto/kotlin/swift_edge_cases.proto [hunk 2]
```diff
@@ -35,8 +35,30 @@ message SwiftEdgeCases {
 
 message Screen {
   oneof screen {
-    string oneof_string = 1;
-    int32 oneof_int32 = 2;
+    string screen_oneof_string = 1;
+    int32 screen_oneof_int32 = 2;
+    SubMessage screen_oneof_sub_message = 3;
```
Comment (dnkoutso) [inline wire-tests/src/commonTest/proto/kotlin/swift_edge_cases.proto:40]: adding this to this edge case proto starts failing compilation

File: wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt [hunk 3]
```diff
@@ -1649,6 +1649,27 @@ class SwiftGenerator private constructor(
...
+    var currentType: ProtoType? = type.type
+
+    while (currentType != null) {
+      val simpleName = currentType.simpleName
+      if (oneOf.name.equals(simpleName, ignoreCase = true)) {
+        val oneOfTypeName = oneOfEnumNames.getValue(oneOf)
+        return oneOfTypeName.peerType("${oneOfTypeName.simpleName}_OneOf")
+      }
+
+      val enclosingName = currentType.enclosingTypeOrPackage ?: break
```
[the remark this answers is not in the document]
Comment (dnkoutso) [reply wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt:1666]: I think its the same, the `continue will run the loop once more but `enclosingName` will be `null` and therefore `currentType` will be `null` and fall to `return oneOfEnumNames.getValue(oneOf)` without the `_OneOf` suffix.
